subsys/bluetooth/mesh/Kconfig

```diff
@@ -21,6 +21,13 @@ if BT_MESH
 config BT_MESH_PROV
 	bool
 
+config BT_MESH_PROVISIONEE
+	bool "Provisionee support"
```

Comment:
We've been using "provisionee" in the Nordic stacks, as it's not ambiguous like "device". I suggested using provisionee in the kconfig entry to align it with the file names, and I think it's the right name for this role, even if it isn't in the spec.

Reply:
I agree about the file name alignment, what I meant was that if we change it in one place then we change it everywhere. Need to think a little bit more about this, since generally I've tried to keep the policy of not introducing terms for features which can't be looked up anywhere in the specs. Maybe it's just because I participated in a lot of the early Mesh WG meetings prior to 1.0 that I got the concept of "device" vs "node" burned into my brain and therefore don't find it ambiguous or confusing at all 😄

Reply:
So will I go back again?

Reply:
@LingaoM no, don't do anything for now

Reply:
@LingaoM @trond-snekvik so is the consensus between you two that you don't want to use the terms "device" or "unprovisioned device" anywhere in the Mesh implementation?

Reply:
Btw, I just took a look at BlueZ, and there we have `prov-initiator.c` and `prov-acceptor.c` (not saying we should copy it, but just collecting up examples of how these are named in various implementations). What other open source mesh stacks are there, or even closed-source with a public API?

Reply:
> We've been using "provisionee" in the Nordic stacks

@trond-snekvik did you have any native English speakers in deciding this? Robin is usually super pedantic when it comes to usage of English language in the specs.

Reply:
..meaning I'm surprised he wouldn't have suggested "provisionee" unless there was some concern with it

Reply:
IMO, in provision, the use of `provisionee` is more appropriate than `device`.

Reply:
> did you have any native English speakers in deciding this?
I don't think so. It's definitely a made up word, so I wouldn't be surprised if the WG torpedoed it.

In other mesh APIs, it's a mix of every possible word:
- Nordic: [Provisionee](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.meshsdk.v4.2.0/group__NRF__MESH__PROV.html?cp=7_2_4_10_1)
- Packetcraft: [Provisioning server](https://github.com/packetcraft-inc/stacks/blob/master/ble-mesh-profile/include/mesh_prv_sr_api.h)
- Silabs: [Provisioning device (PDF!)](https://www.silabs.com/documents/public/reference-manuals/bluetooth-le-and-mesh-software-api-reference-manual.pdf)
-  ESP: [Node](https://github.com/espressif/esp-idf/blob/1c7607d4a67ff96fa532cc4f47838f5ce90b834c/components/bt/esp_ble_mesh/mesh_core/prov.c) (this is a port of the Zephyr stack)
- Cypress: [Provisioning server/node](https://cypresssemiconductorco.github.io/btsdk-docs/BT-SDK/20706-A2_Bluetooth/API/group__wiced__bt__mesh__core.html#gacc7311d144020069d287dc02c6a3a8e1)
- ST Android API: [Target device (PDF!)](https://www.st.com/resource/en/user_manual/dm00434788-bluenrg-mesh-android-api-guide-for-mesh-over-bluetooth-low-energy-stmicroelectronics.pdf)
- Dialog: [Device (PDF!)](https://www.dialog-semiconductor.com/sites/default/files/um-b-098_da1468x_mesh_sdk_rev1v3.pdf)

Only Nordic is using "Provisionee" though, so it's definitely not the conventional name :smile: Perhaps `PROV_DEVICE` is a non-ambiguous middle ground?

Reply:
done

Reply:
prov_dev or prov_device sounds good to me. @LingaoM you might want to reflect that in the c-filenames as well.

Reply:
Done.